### DISCUSSION TITLE
Token for S3 is not refreshed after 1 hour

### DISCUSSION THREAD

**Discussion by HonzaMac** [renovatebot/renovate#26855] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.151.0

### If you're self-hosting Renovate, select which platform you are using.

GitHub Enterprise Server

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Our self-hosted job for scanning repositories start to be running for longer then hour. This leads to token expiration for AWS S3, which is capped on 1 hour. After that tine, we receive warnings in code.

Thank you all for the awesome work on Renovate!

Replication can be done on similar [unit test here](https://github.com/renovatebot/renovate/blob/main/lib/util/cache/repository/impl/s3.spec.ts#L132-L143).

I am willing to work on fix for this. I expect that only proper configuration passed to S3 client is needed.

### Relevant debug logs

<details><summary>Logs</summary>

```
00:19:06   WARN: RepoCacheS3.write() - failure (repository=company/repository-name)
00:19:06         "err": {
00:19:06           "name": "ExpiredToken",
00:19:06           "$fault": "client",
00:19:06           "$metadata": {
00:19:06             "httpStatusCode": 400,
00:19:06             "requestId": "AEN....",
00:19:06             "extendedRequestId": "w...A1E=",
00:19:06             "attempts": 1,
00:19:06             "totalRetryDelay": 0
00:19:06           },
00:19:06           "Code": "ExpiredToken",
00:19:06           "Token-0": "**redacted**",
00:19:06           "RequestId": "AEN....",
00:19:06           "HostId": "w...A1E=",
00:19:06           "message": "The provided token has expired.",
00:19:06           "stack": "ExpiredToken: The provided token has expired.\n    at throwDefaultError (/opt/containerbase/tools/renovate/37.150.1/node_modules/@smithy/smithy-client/dist-cjs/default-error-handler.js:8:22)\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@smithy/smithy-client/dist-cjs/default-error-handler.js:18:39\n    at de_PutObjectCommandError (/opt/containerbase/tools/renovate/37.150.1/node_modules/@aws-sdk/client-s3/dist-cjs/protocols/Aws_restXml.js:5721:12)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@smithy/middleware-serde/dist-cjs/deserializerMiddleware.js:7:24\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@aws-sdk/middleware-signing/dist-cjs/awsAuthMiddleware.js:14:20\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@smithy/middleware-retry/dist-cjs/retryMiddleware.js:27:46\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@aws-sdk/middleware-flexible-checksums/dist-cjs/flexibleChecksumsMiddleware.js:58:20\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26\n    at RepoCacheS3.write (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/util/cache/repository/impl/s3.ts:64:7)\n    at RepoCacheS3.save (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/util/cache/repository/impl/base.ts:85:5)\n    at Object.saveCache (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/util/cache/repository/index.ts:26:5)\n    at finalizeRepo (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/workers/repository/finalize/index.ts:22:3)\n    at Object.renovateRepository (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/workers/repository/index.ts:94:7)\n    at attributes.repository (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/workers/global/index.ts:184:11)\n    at start (/opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/workers/global/index.ts:169:7)\n    at /opt/containerbase/tools/renovate/37.150.1/node_modules/renovate/lib/renovate.ts:18:22"
00:19:06         }
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Where is the 1 hour token generation done? e.g. do you generate it and pass it to Renovate, or does Renovate generate the 1 hour token itself based on some other credentials?

**Reply by HonzaMac** [reply]
I provide credentials as described in documentation. 
I think that AWS client create some temporary access token from provided credentials.

**Reply by rarkins** [reply]
So if you can provide a token which lasts longer than an hour, that would one be one solution. Renovate doesn't have a requirement "only give a token which expires after an hour"

**Reply by HonzaMac** [reply, marked as answer]
I have found the root-cause. 
We provide credentials with "assume-role", which generates new key&secret with limited time. 
We assigned to user policy and stoped using "assume-role" as we can't have credentials longer than 1 hour.

Thank you for help!
